repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -299,6 +304,21 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
       }
     }
 
+    def doUserTokensUpdate(userId: String, authToken: String, ttl: Int): Unit = {
+      if (CacheInvalidationFeedListener.invalidationEnabled()) {
+        //Have to use Vector, because List isn't serializeable in 2.10
+        val oldTokens = Option(datastore.get(s"$USER_ID_KEY_PREFIX$userId").asInstanceOf[Vector[String]]).getOrElse(Vector.empty[String])
+        if (!oldTokens.contains(authToken)) {
+          // @TODO: Do we need to clean up the oldTokens which may have irrelevant values.
+          // @TODO: Nothing is wrong with the data, but it may have extra values that would try to be removed later.
+          val newTokens = oldTokens ++ Vector(authToken)
+          // Updated for later cache invalidation via Atom Feed User events.
+          datastore.remove(s"$USER_ID_KEY_PREFIX$userId")
+          datastore.put(s"$USER_ID_KEY_PREFIX$userId", newTokens, ttl, TimeUnit.SECONDS)
```

Comment:
I agree that this is the standard threading data concurrency issue. Should I move away from storing a serialized Scala Vector in favor of a tokenized String that can be more easily Patched?